source/MissionAction.cpp

```diff
@@ -419,6 +431,25 @@ bool MissionAction::CanBeDone(const PlayerInfo &player, const shared_ptr<Ship> &
 	
 	for(const auto &it : requiredOutfits)
 	{
+		// Special case: map
+		int mapSize = it.first->Get("map");
+		if(mapSize > 0)
```

Comment:
Can you show (in the text-description of the PR) how this would look in a datafile written by some content-writer? (Maybe also for the gift section.) (This usage information is normally done in the "how to use" section from the standard template for PRs.)

Reply:
Sorry about that. I think that I originally reduced the template because it was just a draft and simply wanted to open the subject. When transforming it, I wasn't proposed the template again. I've updated the original post

Reply:
> Some questions from the original post still stand, but as stated they might deserve separate PRs. I'll just keep one of them as current implementation of requiring "local map n" might not be obvious: `HasMapped` currently looks for systems in a specific order, meaning that if you have visited 3 systems and a mission requires you to map only 2, if you didn't visit the _correct_ ones, condition would fail. Do you think we should provide a different method for that?

If this only checks if the player at least knows <nr> systems nearby, then I would expect the check to count all known nearby systems and succeed if the known systems is at least this number (I don't think we want to have a hidden dependency on specific orders).

Reply:
@petervdmeer I think for this "outfits" usage, the mismatch is fine, as it means the player can (or can't) buy the named map outfit in the outfitter. The mission writer isn't able to say `require map <system size> <outfit count>`, so the _mission_ never explicitly states how many systems the player has or hasn't visited, just how many of the map outfit are needed (which should always be 1, 0 or omitted).

Reply:
@tehhowch My concern is more with the "specific order", and "if you didn't visit the correct ones" remarks in the text that I quoted. If that is still an issue, then we would be introducing ambiguity in the keywords that are added here. (I don't like ambiguity for keywords, keywords should be very clearly and exactly defined.)

Reply:
Yeah, my original review missed that this PR is proposing that `require <outfit> <integer>` behave differently when outfit has a "map" attribute vs. not.

That proposal won't be accepted - `<integer>` in a `require` or `outfit` line under an `on ___` action _must_ be the count of that outfit.

Implementing a queryable condition for map knowledge would be where we allow "player knows x of y nearest systems"

Reply:
Yes, currently the existing functions related to map use a specific order.
So I see two possibilities:
* Changing this PR so that `require LocalMap 1` requires a complete map and `require LocalMap 0` requires a non-complete map.
* Making new functions that do the proper checks, by counting visited systems with no defined order.

I feel like writing a separate check would allow for a better result as it allows more interesting flows. I'd even go with percentages so that `mapSize` can vary without actually affecting existing missions. I'd keep `outfit localMap` as it is in that PR: it just "buys" a map as expected.

What do you think could be the next steps?

Reply:
Changing this PR so that `require "Local Map" 1` requires the local map to be complete and `require "Local Map" 0` requires the local map to be incomplete seems to match best with the `give "Local Map"` funcationality also in this PR, so I would recommend to go for this option.

The new functions with the new checks would indeed allow for interesting flows, but [as I said before](https://github.com/endless-sky/endless-sky/pull/5727#issuecomment-827139676), I would suggest to keep those for a separate PR; the new functions will trigger a lot more discussion on desired functionality and exact syntax, while the "Local Map" changes feel more like bug-fixes on keywords already in use (and currently misbehaving).

Reply:
Thank you for the quick reply! I'll try to work this out with tests this week :)

Reply:
Changed behaviour to a binary one as asked:
`require "Local Map" 0`: requires an incomplete map
`require "Local Map" X` where X > 0: requires a complete map